### PR TITLE
REP-2142 Create API Coverage Handler

### PR DESCRIPTION
**Pull request by wdschei**

### CODE REVIEW

File: repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/herp/apivalidatorstatemachinemasked/response-messaging.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (adrianjgeorge) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/herp/apivalidatorstatemachinemasked/response-messaging.cfg.xml]: why do we need a specific response messaging config?

**Comment by adrianjgeorge** [comment]
why is there a herp test on this pull request?

**Comment by wdschei** [comment]
These HERP tests were unknowingly built to test a bug in the API-Checker. As part of the API Converge Handler work this bug was fixed. This caused the HERP tests to fail and so they were updated to account for the bug being fixed.

File: repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/herp/apivalidatorstatemachinemasked/response-messaging.cfg.xml (hunk 1 above)
Comment (wdschei) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/herp/apivalidatorstatemachinemasked/response-messaging.cfg.xml]: Good Question. @tylerroyal Do you know why?

File: repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/apivalidator/statemachine/validator.cfg.xml [hunk 2]
```diff
@@ -6,15 +6,6 @@
...
-                <resource path="/a">
-                    <method name="GET" rax:roles="a:observer a:admin"/>
-                    <method name="POST" rax:roles="a:admin"/>
-                    <method name="DELETE" rax:roles="a:admin"/>
-                </resource>
-            </resources>
-        </application>
-    </validator>
+               dot-output="output.dot"
+               wadl="statemachine.checker"/>
```
Comment (dimtruck) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/apivalidator/statemachine/validator.cfg.xml]: Why is the wadl pointing to statemachine.checker and not the wadl?

Comment (wdschei) [inline repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/filters/apivalidator/statemachine/validator.cfg.xml]: When the API-Checker builds the statemachine.checker from a WADL, often times the assigned ID's are not the same. This makes it impossible to write tests using a WADL.

**Pull request merged**